cyipopt/tests/unit/test_scipy_ipopt_trust_constr.py

```diff
@@ -443,178 +446,97 @@ def hess(x, v):
         return NonlinearConstraint(fun, -np.inf, 0, jac, hess)
 
 
+@pytest.mark.skipif("scipy" not in sys.modules,
+                    reason="Test only valid if Scipy available.")
 class TestTrustRegionConstr(TestCase):
 
     @pytest.mark.slow
     def test_list_of_problems(self):
         list_of_problems = [Maratos(),
-                            Maratos(constr_hess='2-point'),
-                            Maratos(constr_hess=SR1()),
-                            Maratos(constr_jac='2-point', constr_hess=SR1()),
                             MaratosGradInFunc(),
                             HyperbolicIneq(),
-                            HyperbolicIneq(constr_hess='3-point'),
-                            HyperbolicIneq(constr_hess=BFGS()),
-                            HyperbolicIneq(constr_jac='3-point',
-                                           constr_hess=BFGS()),
                             Rosenbrock(),
                             IneqRosenbrock(),
                             EqIneqRosenbrock(),
                             BoundedRosenbrock(),
-                            Elec(n_electrons=2),
-                            Elec(n_electrons=2, constr_hess='2-point'),
-                            Elec(n_electrons=2, constr_hess=SR1()),
-                            Elec(n_electrons=2, constr_jac='3-point',
-                                 constr_hess=SR1())]
+                            Elec(n_electrons=2)]
 
         for prob in list_of_problems:
-            for grad in (prob.grad, '3-point', False):
-                for hess in (prob.hess,
-                             '3-point',
-                             SR1(),
-                             BFGS(exception_strategy='damp_update'),
-                             BFGS(exception_strategy='skip_update')):
-
-                    # Remove exceptions
-                    if grad in ('2-point', '3-point', 'cs', False) and \
-                       hess in ('2-point', '3-point', 'cs'):
-                        continue
-                    if prob.grad is True and grad in ('3-point', False):
-                        continue
-                    with suppress_warnings() as sup:
-                        sup.filter(UserWarning, "delta_grad == 0.0")
-                        result = minimize(prob.fun, prob.x0,
+            for grad in (prob.grad, False):
+                if prob == list_of_problems[1]:  # MaratosGradInFunc
+                    grad = True
+                for hess in (None,):
```

Comment:
We can't use any of the Hessian options now because a) if `hess` is provided, all the constraint Hessians must be provided and b) constraint Hessians can't be transferred from `NonlinearConstraint` to the old constraint format (because the old constraint format in SciPy doesn't ever include constraint Hessians).

It seems possible to get around this, but that would be another PR.